evm-ds/src/cps_executor.rs

```diff
@@ -286,6 +313,41 @@ impl<'a> Handler for CpsExecutor<'a> {
         offset_len: U256,
     ) -> Capture<(ExitReason, Vec<u8>), Self::CallInterrupt> {
         if self.enable_cps {
+            if let Some(result) =
+                self.stack_executor
+                    .precompiles()
+                    .execute(&mut StackExecutorHandle {
+                        executor: &mut self.stack_executor,
+                        code_address,
+                        input: &input,
+                        gas_limit: target_gas,
+                        context: &context,
+                        is_static,
+                    })
+            {
+                return match result {
```

Comment:
Can a precompile itself cause a Capture::Trap? I think it can, e.g. we may have precompiles that actually stop EVM and yield control to Scilla in some non-default way.

Here I don't think we have that.

Reply:
It's possible to do that but it's not implemented yet (deliberately to reduce the scope of changes). We can address that at the time of implementing interop.

Reply:
good, I agree with that approach.  Leaving a TODO might help.